tests/migrations/test_loader.py

```diff
@@ -492,6 +492,22 @@ def test_loading_squashed_ref_squashed(self):
         with self.assertRaisesMessage(NodeNotFoundError, msg):
             loader.graph.forwards_plan(('app1', '3_auto'))
 
+        # Possible if loader.replace_migrations is False.
+        # MigrationExecutor.migration_plan() is responsible for setting it False.
+        loader.replace_migrations = False
+        loader.build_graph()
+        plan = set(loader.graph.forwards_plan(('app1', '3_auto')))
+        plan = plan - loader.applied_migrations.keys()
+        expected_plan = {
+            ('app1', '1_auto'),
+            ('app2', '1_auto'),
+            ('app2', '2_auto'),
+            ('app1', '2_auto'),
+            ('app1', '3_auto'),
+        }
+        self.assertEqual(plan, expected_plan)
+        loader.replace_migrations = True
+
```

Comment:
We didn't make any changes in the loader and this assertions work without a patch. Please move them to the first commit.